crates/re_space_view_time_series/src/line_visualizer_system.rs

```diff
@@ -96,156 +96,193 @@ impl SeriesLineSystem {
     ) -> Result<(), QueryError> {
         re_tracing::profile_function!();
 
-        let query_caches = ctx.entity_db.query_caches();
-        let store = ctx.entity_db.store();
-
         let (plot_bounds, time_per_pixel) = determine_plot_bounds_and_time_per_pixel(ctx, query);
 
-        // TODO(cmc): this should be thread-pooled in case there are a gazillon series in the same plot…
-        for data_result in query.iter_visible_data_results(Self::identifier()) {
-            let annotations = self.annotation_map.find(&data_result.entity_path);
-            let annotation_info = annotations
-                .resolved_class_description(None)
-                .annotation_info();
-            let default_color = DefaultColor::EntityPath(&data_result.entity_path);
-
-            let override_color = lookup_override::<Color>(data_result, ctx).map(|c| c.to_array());
-            let override_series_name = lookup_override::<Name>(data_result, ctx).map(|t| t.0);
-            let override_stroke_width =
-                lookup_override::<StrokeWidth>(data_result, ctx).map(|r| r.0);
-
-            // All the default values for a `PlotPoint`, accounting for both overrides and default
-            // values.
-            let default_point = PlotPoint {
-                time: 0,
-                value: 0.0,
-                attrs: PlotPointAttrs {
-                    label: None,
-                    color: annotation_info.color(override_color, default_color),
-                    marker_size: override_stroke_width.unwrap_or(DEFAULT_STROKE_WIDTH),
-                    kind: PlotSeriesKind::Continuous,
-                },
-            };
-
-            let mut points = Vec::new();
-
-            let time_range = determine_time_range(
-                query,
-                data_result,
-                plot_bounds,
-                ctx.app_options.experimental_plot_query_clamping,
-            );
+        let data_results = query
+            .iter_visible_data_results(Self::identifier())
+            .collect_vec();
```

Comment:
No need to do the collection unconditionally, this can just be part of the parallel case